### PR TITLE
added json api

### PR DESCRIPTION
**Pull request by freester1**
- Added get/post to get email and get all mail for a given email
- Added api_user_table to contain mail_id and mail for json api users
- Added api_user_table to expiration_worker so api_json sessions are temporal
- tests and comments
## GET /api/email 
Sample Request:
```
curl localhost:4000/api/email
```
Sample Response 200:
```
{
  "session_id":"0ymDyDqirJ1D2ammhQWx3tuGAVX",
  "email":"0ymDyCIwnvAscVllznRlFXCjm1e@mailchan.moe"
}
```
## POST /api/mail/$email
Sample Request:
```
curl -H 'Content-Type: application/json' localhost:4000/api/0ymDyCIwnvAscVllznRlFXCjm1e@mailchan.moe -X POST -d '{"session_id": "0ymDyDqirJ1D2ammhQWx3tuGAVX"}'
```
Sample Response 200:
```
{"emails":[]}
```
or with mail:
```
{"emails":[{"to":"bar","subject":null,"from":"baz","date":null,"body":null}]}
```
Returns 404 when:
- session_id could not be found or is not provided
- the email could not be found

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -8,7 +8,7 @@ use Mix.Config
 # with brunch.io to recompile .js and .css sources.
 config :mailchan, MailchanWeb.Endpoint,
   http: [port: 4000],
-  debug_errors: true,
```
Comment (Ninja3047) [inline config/dev.exs]: Why did you change it to false in dev mode 🤔

Comment (freester1) [reply config/dev.exs]: when it is true then on  an error phoenix will send down an html page containing info about the error. It can be useful but I mainly just use iex/curl to test so i can't render the html anyways

**Comment by Ninja3047** [comment]
🍕
